src/lib/components/AssistantSettings.svelte

```diff
@@ -46,9 +47,9 @@
 		compress = module.readAndCompressImage;
 
 		if (assistant) {
-			modelId = assistant.modelId;
+			modelId = findCurrentModel(models, assistant.modelId).id;
 		} else {
-			modelId = models.find((model) => model.id === $settings.activeModel)?.id ?? models[0].id;
+			modelId = findCurrentModel(models, $settings.activeModel).id;
 		}
```

Comment:
```suggestion
		const modelId = findCurrentModel(models, assistant ? assistant.modelId : $settings.activeModel).id;
```

didn't test. but I think we can do it like this